index.js

```diff
@@ -172,11 +172,19 @@ function shouldRemove(node) {
         return options['remove-comments'] || isEmpty(node);
     }
 
-    if (options['remove-empty-tags'].indexOf(node.name) != -1) {
+    if (isListedInOptions('remove-empty-tags', node.name)) {
         return isEmpty(node);
     }
 
-    return options['remove-tags'].indexOf(node.name) != -1;
+    return isListedInOptions('remove-tags', node.name);
+}
+
+function isListedInOptions(optionsArrayName, name) {
+    var matches = options[optionsArrayName].filter(function(option) {
+        return typeof option !== 'string' && option.test(name) || option === name;
```

Comment:
Why not `option instanceof RegExp`?

Reply:
good question. It does not really make that big difference here though... If you think it is important I can of course adapt it

Reply:
I think that would be better. If it's not a string, `option.test` might throw an exception. Not that anything else is supported, but it seems more correct.